request: x265 with SVT-HEVC

x265 now allows to use SVT-HEVC (see: https://x265.readthedocs.io/en/latest/svthevc.html).
(no clue why they didn't make a separate project out of this)
Still it would be nice if media-autobuild_suite would also support this.

not sure whether it help, I enabled STV-HEVC and for me it stops at:
```
16:08:57   Running git update for SVT-HEVC...
16:08:58 ┌ SVT-HEVC git ................................ [Files missing]
16:09:04 ├ Running cmake...
16:09:13 ├ Running build...
Likely error:
                 from ../Source/Lib/C_DEFAULT/EbSampleAdaptiveOffset_C.c:7:
I:/media-autobuild_suite/msys64/mingw32/lib/gcc/i686-w64-mingw32/9.1.0/include/emmintrin.h:143:1: error: inlining failed in call to always_inline '_mm_load_sd': target specific option mismatch
  143 | _mm_load_sd (double const *__P)
      | ^~~~~~~~~~~
In file included from ../Source/Lib/C_DEFAULT/EbSampleAdaptiveOffset_C.h:9,
                 from ../Source/Lib/C_DEFAULT/EbSampleAdaptiveOffset_C.c:7:
../Source/Lib/Codec/EbDefinitions.h:266:9: note: called from here
  266 |         _mm_store_sd((double*)(dst + i), _mm_load_sd((const double*)(src + i)));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
build failed. Check I:/media-autobuild_suite/build/SVT-HEVC-git/build-32bit/ab-suite.build.log
This is required for other packages, so this script will exit.
16:09:16   Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/ze4n.zip
Copy and paste [logs.zip](https://0x0.st/ze4n.zip) in the GitHub issue.
```

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1362,13 +1362,11 @@ if [[ ! $x265 = "n" ]] && do_vcs "hg::https://bitbucket.org/multicoreware/x265";
     do_patch "https://gist.githubusercontent.com/1480c1/7743056f52e4a546294e2160431674bc/raw/0001-CMake-alias-CMAKE_INSTALL_BINDIR-to-BIN_INSTALL_DIR.patch"
 
     do_x265_cmake() {
-        do_print_progress "Building $1" && shift 1
-        extra_script pre cmake
-        log "cmake" cmake "$LOCALBUILDDIR/$(get_first_subdir)/source" -G Ninja \
-        -DCMAKE_INSTALL_PREFIX="$LOCALDESTDIR" -DBIN_INSTALL_DIR="$LOCALDESTDIR/bin-video" \
+        do_print_progress "Building $1" && shift
+        do_cmake video "$(realpath --relative-to="$PWD" "$LOCALBUILDDIR/$(get_first_subdir)/source")" \
+        skip_build_dir
         -DENABLE_SHARED=OFF -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=ON -DHG_EXECUTABLE=/usr/bin/hg.bat \
         -DENABLE_HDR10_PLUS=ON $xpsupport "$@"
-        extra_script post cmake
         do_ninja
     }
     build_x265() {

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1359,6 +1359,7 @@ if [[ ! $x265 = "n" ]] && do_vcs "hg::https://bitbucket.org/multicoreware/x265";
     do_uninstall libx265{_main10,_main12}.a bin-video/libx265_main{10,12}.dll "${_check[@]}"
     [[ $bits = "32bit" ]] && assembly="-DENABLE_ASSEMBLY=OFF"
     [[ $x265 = d ]] && xpsupport="-DWINXP_SUPPORT=ON"
+    do_patch "https://gist.githubusercontent.com/1480c1/7743056f52e4a546294e2160431674bc/raw/0001-CMake-alias-CMAKE_INSTALL_BINDIR-to-BIN_INSTALL_DIR.patch"
 
     build_x265() {
         create_build_dir

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1361,6 +1361,12 @@ if [[ ! $x265 = "n" ]] && do_vcs "hg::https://bitbucket.org/multicoreware/x265";
     [[ $x265 = d ]] && xpsupport="-DWINXP_SUPPORT=ON"
     do_patch "https://gist.githubusercontent.com/1480c1/7743056f52e4a546294e2160431674bc/raw/0001-CMake-alias-CMAKE_INSTALL_BINDIR-to-BIN_INSTALL_DIR.patch"
 
+    if [[ $svthevc = "y" ]]; then
+        export SVT_HEVC_INCLUDE_DIR="$LOCALDESTDIR/include/svt-hevc"
+        export SVT_HEVC_LIBRARY_DIR="$LOCALDESTDIR/lib"
+        x265_cmake_extra+=(-DENABLE_SVT_HEVC=ON)
+    fi
+
     do_x265_cmake() {
         do_print_progress "Building $1" && shift
         do_cmake video "$(realpath --relative-to="$PWD" "$LOCALBUILDDIR/$(get_first_subdir)/source")" \

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1229,7 +1229,6 @@ if [[ $bits = "32bit" ]]; then
 elif { [[ $svthevc = "y" ]] || enabled libsvthevc; } &&
     do_vcs "https://github.com/OpenVisualCloud/SVT-HEVC.git"; then
     do_uninstall "${_check[@]}" include/svt-hevc
-    do_patch "https://github.com/OpenVisualCloud/SVT-HEVC/pull/206.patch" am
     do_cmakeinstall video -DUNIX=OFF -DNATIVE=OFF
     do_checkIfExist
 fi

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1426,7 +1426,7 @@ EOF
     fi
     }
     build_x265
-    cpuCount=1 log "install" ninja install
+    do_ninjainstall
     if [[ $standalone = y && $x265 = d ]]; then
         cd_safe "${LOCALBUILDDIR}/$(get_first_subdir)"
         do_uninstall bin-video/x265-numa.exe

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1440,7 +1440,7 @@ EOF
         _check+=(bin-video/x265-numa.exe)
     fi
     do_checkIfExist
-    unset xpsupport assembly cli
+    unset xpsupport assembly cli x265_cmake_extra
 else
     pc_exists x265 || do_removeOption "--enable-libx265"
 fi

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1369,10 +1369,11 @@ if [[ ! $x265 = "n" ]] && do_vcs "hg::https://bitbucket.org/multicoreware/x265";
 
     do_x265_cmake() {
         do_print_progress "Building $1" && shift
-        do_cmake video "$(realpath --relative-to="$PWD" "$LOCALBUILDDIR/$(get_first_subdir)/source")" \
-        skip_build_dir
-        -DENABLE_SHARED=OFF -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=ON -DHG_EXECUTABLE=/usr/bin/hg.bat \
-        -DENABLE_HDR10_PLUS=ON $xpsupport "$@"
+        do_cmake video skip_build_dir \
+            "$(realpath --relative-to="$PWD" "$LOCALBUILDDIR/$(get_first_subdir)/source")" \
+            -DENABLE_SHARED=OFF -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=ON \
+            -DHG_EXECUTABLE=/usr/bin/hg.bat -DENABLE_HDR10_PLUS=ON $xpsupport \
+            ${x265_cmake_extra[@]} "$@"
         do_ninja
     }
     build_x265() {

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1361,12 +1361,6 @@ if [[ ! $x265 = "n" ]] && do_vcs "hg::https://bitbucket.org/multicoreware/x265";
     [[ $x265 = d ]] && xpsupport="-DWINXP_SUPPORT=ON"
     do_patch "https://gist.githubusercontent.com/1480c1/7743056f52e4a546294e2160431674bc/raw/0001-CMake-alias-CMAKE_INSTALL_BINDIR-to-BIN_INSTALL_DIR.patch"
 
-    build_x265() {
-        create_build_dir
-        local build_root
-        build_root="$(pwd)"
-        mkdir -p {8,10,12}bit
-
     do_x265_cmake() {
         do_print_progress "Building $1" && shift 1
         extra_script pre cmake
@@ -1377,6 +1371,10 @@ if [[ ! $x265 = "n" ]] && do_vcs "hg::https://bitbucket.org/multicoreware/x265";
         extra_script post cmake
         do_ninja
     }
+    build_x265() {
+        create_build_dir
+        local build_root="$(pwd)"
+        mkdir -p {8,10,12}bit
     [[ $standalone = y ]] && cli="-DENABLE_CLI=ON"
 
     if [[ $x265 =~ (o12|s|d|y) ]]; then

